### ISSUE TITLE
False positive with __all__ and submodules

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#126]
Originally reported by: **Boris FELD (BitBucket: [lothiraldan](http://bitbucket.org/lothiraldan), GitHub: @lothiraldan?)**

---

Hi I just get a bug report by mail:

```
I believe the new E0603 error is incorrectly flagging valid code. In particular when __all__ is part of the __init__.py file of a package the contents can refer to sub-modules of the package i.e.,

mypkg/
   __init__.py
   one.py
   two.py

if __init__.py has this:

__all__ = [ "one" ]

then the affect of "from mypkg import *" is to import sub-module one, but not sub-module two. pylint is now flagging this as an error b/c "one" is not a variable defined within __init__.py

What's worse is that if you specify E0603 in the disable line of an .rc config file, it is ignored and the error is still printed.
```

I wasn't aware of this behavior of **all** and submodules. I think it will be difficult to fix because we do analysis per module for this error if I remember correctly and in order to fix the false positive, we must be able to check or defer check we we have the list of all modules and submodules.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/126
